modules/pubmaticAnalyticsAdapter.js

```diff
@@ -429,7 +429,7 @@ function getFloorFetchStatus(floorData) {
 function getListOfIdentityPartners() {
   const namespace = getGlobal();
   const publisherProvidedEids = namespace.getConfig("ortb2.user.eids") || [];
-  const availableUserIds = namespace.adUnits[0]?.bids[0]?.userId || {};
+  const availableUserIds = namespace.getUserIds() || {};
```

Comment:
A few remarks:

 - the publisher can provide EIDs in a number of other ways beside globally set `ortb2.user.eids`. They could be in `ortb2.user.ext.eids`, and they could be in auction-specific and/or bidder-specific first party data.
 - prebid normalizes all EIDs (from publisher and from userId modules, from both ortb2 locations, both bidder-specific and not) into each bidderRequest's `ortb2.user.ext.eids`. Auction objects also have a `getFPD()` method that allows extracting only those that are not bidder-specific (`getFPD().global.user.ext.eids`).
 - the keys in the map returned by `getUserIds` are arbitrary labels chosen by userId modules, not necessarily related to how they're stored (i'm referring to lines 434-439).